spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -112,9 +115,22 @@ public void setTypeMapper(Jackson2JavaTypeMapper typeMapper) {
 		this.typeMapper = typeMapper;
 	}
 
+	/**
+	 * Configure the JsonDeserializer to use key type headers.
+	 * @param isKey Use key type headers if true
+	 */
+	public void configureTypeMapper(boolean isKey) {
```

Comment:
How about instead of those new ctors and extra arg there just rename this to the setter - `setUseTypeMapperForKey(boolean)`  and encourage end-users to call it for the particular reason in the class JavaDocs?

And right: still call it from the `configure()`.

How is this approach with you?

We will have a new functionality and won't have any breaking changes.

I'm open form any other suggestions.
But I'm against the breaking change to the `JsonDeserializer(Class<T> targetType, ObjectMapper objectMapper)` ctor. We can't afford it in the current point release.

Reply:
That sounds better, thanks - do you think the ```JsonSerde``` constructor should still have an ```isKey``` param, in addition to the original ctors?

In order to configure streams like this:
```
.stream(OBJECT_INPUT_TOPIC, Consumed.with(
new JsonSerde<>(JsonObjectKey.class, true), 
new JsonSerde<>(JsonObjectValue.class)));
```
So I think the jsonSerde ctor is ok?

It would be great is spring just intercepted the serde and didn't need a constructor arg though ;)

Reply:
😄 OK. Agreed. Only the concern to keep backward compatibility.